it/controllers/ListYearsFeatureSwitchISpec.scala

```diff
@@ -157,4 +195,60 @@ class ListYearsFeatureSwitchISpec extends ISpecBase {
     Jsoup.parse(res.body)
   }
 
+  private def postIsCorrectAgentPage(scenario: String) = {
+
+    implicit val hc: HeaderCarrier = HeaderCarrier(sessionId = Some(SessionId(testSessionId)))
+
+    lazy val mockAppointAgentSessionRepository: AppointAgentSessionRepository = app.injector.instanceOf[AppointAgentSessionRepository]
+
+    val searchedAgentData: SearchedAgent = SearchedAgent.apply(1001, "Agent", "Address", AgentSelected, None)
+
+    val selectedAgentData = SelectedAgent.apply(searchedAgentData, isTheCorrectAgent = true, None, None)
+
+    await(mockAppointAgentSessionRepository.saveOrUpdate(selectedAgentData))
+
+    stubFor {
+      get("/business-rates-authorisation/authenticate")
+        .willReturn {
+          aResponse.withStatus(OK).withBody(Json.toJson(testAccounts).toString())
+        }
+    }
+
+    stubFor {
+      post("/auth/authorise")
+        .willReturn {
+          aResponse.withStatus(OK).withBody("{}")
+        }
+    }
+
+    val authData: OwnerAuthResult = scenario match {
+      case "OneAuthOneAgent" => testOwnerAuthResult1
+      case "NoAuths" => testOwnerAuthResultNoProperties
+      case "OneAuthNoAgent" => ownerAuthResultWithOneAuthorisation
```

Comment:
`case "OneAuthNoAgent" => ownerAuthResultWithOneAuthorisation`


`ownerAuthResultWithOneAuthorisation` has an agent but is testing when there are no agents

```  val ownerAuthResultWithOneAuthorisation = OwnerAuthResult(
    start = 1,
    size = 15,
    filterTotal = 1,
    total = 1,
    authorisations = Seq(OwnerAuthorisation(
      authorisationId = 4222211L,
      status = "APPROVED",
      submissionId = "PLSubId",
      uarn = 999000111L,
      address = "123, SOME ADDRESS, SOME TOWN, AB1 CD2",
      localAuthorityRef = "BAREF",
      agents = Seq(OwnerAuthAgent(
        authorisedPartyId = 12L,
        organisationId = 1L,
        organisationName = "Some Org name",
        agentCode = 1001
      ))
    ))
  )```